frontend/src/lib/validation.ts

```
@@ -0,0 +1,294 @@
+  input: string,
+  maxLength: number = 1000
+): string {
+  if (typeof input !== "string") {
+    return "";
+  }
+
+  return input
+    .trim()
+    .slice(0, maxLength)
+    .replace(/[<>\"'&]/g, "") // Remove potentially dangerous HTML characters
+    .replace(/[\x00-\x1f\x7f-\x9f]/g, ""); // Remove control characters
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)